data-access/mandataris-download.ts

```diff
@@ -0,0 +1,257 @@
+import { query, sparqlEscapeString } from 'mu';
+import { getSparqlResults } from '../util/sparql-result';
+import { sparqlEscapeDateTime, sparqlEscapeUri } from '../util/mu';
+
+export const downloadMandatarissen = {
+  getWithFilters,
+  getPropertiesOfMandatarissen,
+};
+
+async function getWithFilters(filters) {
+  const {
+    bestuursperiodeId,
+    bestuursorgaanId,
+    persoonIds,
+    bestuursFunctieCodeIds,
+  } = filters;
+  let bestuursorgaanInTijdFilter: string | null = null;
+  let onlyActiveFilter: string | null = null;
+  let persoonFilter: string | null = null;
+  let mandaatTypeFilter: string | null = null;
+
+  if (bestuursorgaanId) {
+    bestuursorgaanInTijdFilter = `
+      ?bestuursorgaanInTijd org:hasPost ?mandaat.
+      ?bestuursorgaanInTijd mu:uuid ${sparqlEscapeString(bestuursorgaanId)}.
+    `;
+  }
+  if (filters.activeOnly) {
+    const escapedTodaysDate = sparqlEscapeDateTime(new Date());
+    onlyActiveFilter = `
+      ?mandataris mandaat:einde ?einde.
+
+      ?bestuursorgaanInTijd mandaat:bindingStart ?startBestuursorgaan. 
+      ?bestuursorgaanInTijd mandaat:bindingEinde ?eindeBestuursorgaan. 
+
+      FILTER (
+        ?einde <= ?eindeBestuursorgaan &&
+        ?einde >= ?startBestuursorgaan &&
+        ?einde >= ${escapedTodaysDate}
+      )
+      BIND(IF(BOUND(?eindeBestuursorgaan), ?eindeBestuursorgaan, ${escapedTodaysDate}) AS ?saveEindeBestuursorgaan).
```

Comment:
saFe

Reply:
this won't be enough though because the mandataris end date AND the boit bindingeinde should be optional. This way you're filtering out all mandataris instances where there is no enddate set for the bestuursorgaan or the mandataris

Reply:
by optional i really mean an OPTIONAL statement in the query right?

Reply:
When looking at the code I didn;t even use the ?saveBestuursorgaanEinde variable..

Updated it with the optional